orchestra/gitutils/__init__.py

```diff
@@ -86,17 +81,22 @@ def current_branch_info(repo_path):
     if branch_file.is_file():
         commit = branch_file.read_text().strip()
     else:
-        # Look into .git/info/refs
-        refs = (dot_git / "info" / "refs").read_text().strip()
-        commit = _only(
-            [
-                match[1]
-                for match in [
-                    re.match(rf"^([0-9a-f]*)\s+refs/heads/{branch}$", line.strip()) for line in refs.split("\n")
-                ]
-                if match
+        # Look into .git/info/refs and into .git/packed-refs, if they are files.
+        if (dot_git / "info" / "refs").is_file():
+            refs = (dot_git / "info" / "refs").read_text().strip()
+        if (dot_git / "packed-refs").is_file():
+            refs += (dot_git / "packed-refs").read_text().strip()
+
+        matching_lines = [
+            match[1]
+            for match in [
+                re.match(rf"^([0-9a-f]*)\s+refs/heads/{branch}$", line.strip()) for line in refs.split("\n")
             ]
-        )
+            if match
+        ]
+        assert len(matching_lines) == 1, f"Expected exactly one match for branch '{branch}', got: {len(matching_lines)}"
+        assert all(matching_lines[0] == x for x in matching_lines), f"Different matches for branch '{branch}'"
```

Comment:
These two asserts are redundant, either:
* We say that there can only be one matching line (first assert)
* We say that there can be the same line in multiple places (second assert)

I'm not 100% sure about the git internals, but it seems that `.git/info/refs` and `.git/packed-refs` are two independent mechanisms, so there might be the situation in which are both present with a line, I'd change the `== 1` to a `>= 1`

Reply:
Right.

Reply:
Ok, take a look now!

Reply:
:+1: